### PR TITLE
Run notebook tests instead of IPython tests in Dockerfile

### LINKED ISSUE #971: Tests fail on new builds of Jupyter Notebook
The tests run during the build of Jupyter Notebook on Docker Hub fail, please see the [full build log](https://hub.docker.com/r/jupyter/notebook/builds/bys4ac8vttnqujuajmdygkk/). This issue arised today and is most likely due to a broken dependency on a 3rd package: the 4.0.0 and 4.1.0 release all show the same issue.

### PR DESCRIPTION
**Pull request by minrk**
IPython tests are failing due to upstream changes in pygments (not a bug, just a test result change)

closes #971

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -84,13 +85,11 @@ RUN BUILD_DEPS="nodejs-legacy npm" && \
...
-RUN pip2 install --no-cache-dir mock nose requests testpath && \
-    pip3 install --no-cache-dir nose requests testpath && \
+RUN pip3 install --no-cache-dir notebook[test] && \
     \
-    iptest2 && iptest3 && \
+    nosetests notebook && \
     \
-    pip2 uninstall -y funcsigs mock nose pbr requests six testpath && \
-    pip3 uninstall -y nose requests testpath
+    pip3 uninstall -y nose requests
```
Comment (takluyver) [inline Dockerfile]: Presumably if notebook or any of its dependencies added a dependency on `requests`, this would be broken until we changed this line?

Comment (minrk) [inline Dockerfile]: yup. This is part of the strict cleanup stuff that's been added over time. I don't like any of it. I'll remove this step.

**Comment by jdfreder** [comment]
Thanks for looking into this, I'm assuming this is what's causing the failures in #931

**Comment by minrk** [comment]
@jdfreder this hasn't anything to do with what you are seeing in #931. The Dockerfile isn't relevant to those tests, which are failing due to real bugs.

**Comment by jdfreder** [comment]
> @jdfreder this hasn't anything to do with what you are seeing in #931. The Dockerfile isn't relevant to those tests, which are failing due to real bugs.

Darn, wishful thinking...

**Pull request merged**